### DISCUSSION TITLE
What isn't upstream?

### DISCUSSION THREAD

**Discussion by scythe000** [linux-surface/linux-surface#1441] [General]
Have most of these changes made it upstream yet? I see that Tails still won't recognize a Surface Laptop Studio's touchpad or keyboard.

**Comment by qzed** [comment]
Support for the Surface Laptop Studio 1 keyboard and touchpad is supported upstream. However, whether it's available in a distro depends on that distro's kernel version and config. For the SLS1, any kernel over 5.16 should do.

The Surface Laptop Studio 2 is still a work in progress, however.

**Comment by scythe000** [comment]
I think from what I’m reading, currently all functions are supported, no? At least by linux-surface.

- Sent from my iPhone. I often use speech to text transcription; please excuse any grammatical or spelling errors. -

On May 7, 2024, at 1:56 PM, Maximilian Luz ***@***.***> wrote:

﻿

Support for the Surface Laptop Studio 1 keyboard and touchpad is supported upstream. However, whether it's available in a distro depends on that distro's kernel version and config. For the SLS1, any kernel over 5.16 should do.

The Surface Laptop Studio 2 is still a work in progress, however.
